vault/kv_helpers.go

```diff
@@ -78,6 +78,27 @@ func kvReadRequest(client *api.Client, path string, params map[string]string) (*
 	return api.ParseSecret(resp.Body)
 }
 
+func kvListRequest(client *api.Client, path string) ([]interface{}, error) {
+	log.Printf("[DEBUG] Listing secrets at %s from Vault", path)
+	resp, err := client.Logical().List(path)
+	if err != nil {
+		return nil, fmt.Errorf("error listing from Vault at path %s, err=%s", path, err)
+	}
+
+	if resp == nil {
+		return nil, fmt.Errorf("no secrets found at %q", path)
+	}
+
+	// Return key names if they are present in response
+	if keyNameList, ok := resp.Data["keys"]; ok && keyNameList != nil {
+		if keyNames, ok := keyNameList.([]interface{}); ok {
+			return keyNames, nil
+		}
+	}
+
+	return nil, fmt.Errorf("keys are either not present or incorrectly formatted in response from Vault")
```

Comment:
Ah, you can move this error to the type asset check above, and make the message explicit: `expected %T`